apm-sniffer/apm-sdk-plugin/influxdb-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/influxdb/define/InfluxDBInstrumentation.java

```diff
@@ -0,0 +1,91 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.influxdb.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+import org.apache.skywalking.apm.plugin.influxdb.InfluxDBMethodMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+/**
+ * enhance InfluxDB InfluxDBFactory
+ * really impl class {@link org.influxdb.impl.InfluxDBImpl}
+ *
+ * @since  2020/05/22
+ */
+public class InfluxDBInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "org.influxdb.impl.InfluxDBImpl";
+    private static final String INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.influxdb.interceptor.InfluxDBConstructorInterceptor";
+  private static final String INFLUXDB_METHOD_INTERCEPT_CLASS = "org.apache.skywalking.apm.plugin.influxdb.interceptor.InfluxDBMethodInterceptor";
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return NameMatch.byName(ENHANCE_CLASS);
+    }
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[] {
+            new ConstructorInterceptPoint() {
+                @Override
+                public ElementMatcher<MethodDescription> getConstructorMatcher() {
+                    return takesArgument(0, String.class);
+                }
+
+                @Override
+                public String getConstructorInterceptor() {
+                    return INTERCEPTOR_CLASS;
+                }
+            }
+        };
+    }
+
+    @Override
+    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+      return new InstanceMethodsInterceptPoint[] {
+          new InstanceMethodsInterceptPoint() {
+            @Override
+            public ElementMatcher<MethodDescription> getMethodsMatcher() {
+              return InfluxDBMethodMatch.INSTANCE.getInfluxDBMethodMatcher();
```

Comment:
This is good for encapsulation, but it's a pitfall that we check third-party classes in classes whose name end with `*Instrumentation`

https://github.com/apache/skywalking/blob/f3d907bf6df924e7cc7e65efbdfd847920ce47e5/apm-checkstyle/checkStyle.xml#L122-L129

to avoid issues like this https://github.com/apache/skywalking/pull/2871 , but this breaks the checks, although there is no third-party class in the `InfluxDBMethodMatch` for now, other reviewers should pay attention to this class in the future, FYI @wu-sheng

Reply:
Question, why need this singleton. This method is not called in high frequently, and nothing cached there.

Reply:
I did it the other way